### DISCUSSION TITLE
Error: setState() called after dispose()

### DISCUSSION THREAD

**Discussion by Schrolli91** [danvick/flutter_form_builder#1318] [General]
I have a little problem with the following code:

```dart
FormBuilderField(
  initialValue: goods?.image ?? 'placeholder',
  name: 'image',
  builder: (field) {
    return GestureDetector(
      onTap: () async {
        String? imageName =
            await context.pushNamed('image_selector');
        print(imageName);
        field.didChange(imageName);
      },
      child: Image.asset(
        'assets/food_icons/${formKey.currentState?.fields['image']?.value ?? "placeholder"}.png',
        height: 70,
      ),
    );
  },
),
```

This will work as expected if i add a new item (`goods` is null).
But if i want to edit one (`goods` is not null, so all fields a prefilled like `goods?.image ?? 'placeholder'`) then i will get the `setState called after` Error.

But for me its not clear why this happens. Any Ideas?

![image](https://github.com/flutter-form-builder-ecosystem/flutter_form_builder/assets/6043382/f37b22c9-6fb6-4c0e-a7d2-9cb06adac95d)

**Comment by deandreamatias** [comment]
This still an issue? @Schrolli91
